src/client/app/components/LanguageSelectorComponent.tsx

```diff
@@ -3,97 +3,52 @@
+ * @returns Language selector element for navbar
  */
-export default class LanguageSelectorComponent extends React.Component<LanguageSelectProps, DropdownState> {
-	constructor(props: LanguageSelectProps) {
-		super(props);
-		this.handleChangeLanguage = this.handleChangeLanguage.bind(this);
-		this.toggleDropdown = this.toggleDropdown.bind(this);
-		this.state = {
-			dropdownOpen: false,
-			compareSortingDropdownOpen: false
-		};
-	}
-
-	public render() {
-		const divBottomPadding: React.CSSProperties = {
-			paddingBottom: '15px'
-		};
-
-		const labelStyle: React.CSSProperties = {
-			fontWeight: 'bold',
-			margin: 0
-		};
-
-		return (
-			<div style={divBottomPadding}>
-				<p style={labelStyle}>
-					<FormattedMessage id='language' />:
-				</p>
-				<Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggleDropdown}>
-					<DropdownToggle outline caret>
-						{/* Show the currently selected language as its name */}
-						{LanguageNames[this.props.selectedLanguage]}
-					</DropdownToggle>
-					<DropdownMenu>
-						<DropdownItem
-							onClick={() => this.handleChangeLanguage(LanguageTypes.en)}
-						>
-							English
-						</DropdownItem>
-						<DropdownItem
-							onClick={() => this.handleChangeLanguage(LanguageTypes.fr)}
-						>
-							Français
-						</DropdownItem>
-						<DropdownItem
-							onClick={() => this.handleChangeLanguage(LanguageTypes.es)}
-						>
-							Español
-						</DropdownItem>
-					</DropdownMenu>
-				</Dropdown>
-				<div>
-					<TooltipMarkerComponent page='home' helpTextId='help.home.language' />
-				</div>
-			</div>
-		);
-	}
-
-	private handleChangeLanguage(value: LanguageTypes) {
-		this.props.changeLanguage(value);
-	}
-
-	private toggleDropdown() {
-		this.setState(prevState => ({ dropdownOpen: !prevState.dropdownOpen }));
-	}
+export default function LanguageSelectorComponent() {
+	const dispatch = useDispatch();
+
+	const selectedLanguage = useSelector((state: State) => state.options.selectedLanguage);
+	const version = useSelector((state: State) => state.version.version);
+
+	return (
+		<>
+			<UncontrolledDropdown direction='start'>
+				<DropdownToggle nav caret>
+					<FormattedMessage id='language' />
+				</DropdownToggle>
+				<DropdownMenu>
+					<DropdownItem
+						onClick={() => dispatch(updateSelectedLanguage(LanguageTypes.en))}
+						disabled={selectedLanguage === LanguageTypes.en}>
+						English
+					</DropdownItem>
+					<DropdownItem
+						onClick={() => dispatch(updateSelectedLanguage(LanguageTypes.fr))}
+						disabled={selectedLanguage === LanguageTypes.fr}>
+						Français
+					</DropdownItem>
+					<DropdownItem
+						onClick={() => dispatch(updateSelectedLanguage(LanguageTypes.es))}
+						disabled={selectedLanguage === LanguageTypes.es}>
+						Español
+					</DropdownItem>
+					<DropdownItem divider />
+					<DropdownItem
+						href={'https://openenergydashboard.github.io/help/' + version + '/language.html'}>
```

Comment:
That is a good idea, I meant to make note of this but never wrote it down anywhere.

Reply:
Adding to this - if we do add support for admins to host their own help pages, then I think it makes sense to make that URL  either a .env variable or an admin preferences setting, instead of compiled into the code directly. Should an issue be opened that addresses this?

Reply:
Feel free to add an issue if you want. I was waiting until after the admin page changes were complete to consider doing this. We don't yet have a request for this but it was an idea.